core/primitives/src/views.rs

```diff
@@ -363,12 +363,28 @@ pub struct KnownProducerView {
     pub next_hops: Option<Vec<PublicKey>>,
 }
 
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct Tier1ProxyView {
+    pub addr: std::net::SocketAddr,
+    pub peer_id: PublicKey,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct AccountDataView {
+    pub peer_id: PublicKey,
```

Comment:
can you have both peer_id and proxies set?

Reply:
You have to. peer_id is used for routing through proxies. Also I intend to deprecate AnnounceAccount in favor of peer_id in AccountData, so it will be used for routing over TIER2 as well.